Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,747 @@
+                  controllerId = [attr stringValue];
+                }
+
+              if ([controllerPlaceholders count] > 0)
+                {
+                  NSXMLElement *ce = [controllerPlaceholders objectAtIndex: 0];
+                  NSXMLNode *attr = [ce attributeForName: @"id"];
+                  controllerId = [attr stringValue];
+                }
+
+              NSArray *customObjects = [document nodesForXPath: @"//objects/customObject" error: NULL];
+              NSEnumerator *coen = [customObjects objectEnumerator];
+              NSXMLElement *coel = nil;
+              while ((coel = [coen nextObject]) != nil)
+                {
+                   NSXMLNode *attr = [coel attributeForName: @"sceneMemberID"];
+                  if ([[attr stringValue] isEqualToString: @"firstResponder"])
+                    {
+                      NSXMLNode *customClassAttr = [coel attributeForName: @"customClass"];
+                      NSXMLNode *idAttr = [coel attributeForName: @"id"];
+                      NSString *originalId = [idAttr stringValue];
+
+                      [idAttr setStringValue: @"-1"]; // set to first responder id
+                      [customClassAttr setStringValue: @"FirstResponder"];
+
+                      // Actions
+                      NSArray *cons = [document nodesForXPath: @"//action" error: NULL];
+                      NSEnumerator *consen = [cons objectEnumerator];
+                      NSXMLElement *celem = nil;
+
+                      while ((celem = [consen nextObject]) != nil)
+                        {
+                          NSXMLNode *targetAttr = [celem attributeForName: @"target"];
+                          NSString *val = [targetAttr stringValue];
+                          if ([val isEqualToString: originalId])
+                            {
+                              [targetAttr setStringValue: @"-1"];
+                            }
+                        }
+
+                      // Outlets
+                      cons = [document nodesForXPath: @"//outlet" error: NULL];
+                      consen = [cons objectEnumerator];
+                      celem = nil;
+                      while ((celem = [consen nextObject]) != nil)
+                        {
+                          NSXMLNode *attr = [celem attributeForName: @"destination"];
+                          NSString *val = [attr stringValue];
+                          if ([val isEqualToString: originalId])
+                            {
+                              [attr setStringValue: @"-1"];
+                            }
+                        }
+                    }
+                }
+
+              // Create document...
+              [_scenesMap setObject: document
+                             forKey: sceneId];
+              
+              // Map controllerId's to scenes...
+              if (controllerId != nil)
+                {
+                  [_controllerMap setObject: sceneId
```

Comment:
You could call processSegues:forId: here already and store the result of that in a map.

Reply:
Code has been restructured.

Reply:
All the restructuring is good but it does not address the point I am making. What I was suggesting was to do all the XML transformations here. That way we would already store the results (Maybe even in just one map not with two) and the code later would just read from that map.

Reply:
The changes work.  They are efficient.  Why put all of the transformations in one place?

Reply:
The changes work.  They are efficient.  Why put all of the transformations in one place?

Reply:
Sorry, we seem to have different goals here. You seem to want code that is basically working. Working code is important but for me this is where it starts. What we need is code that is maintainable. The minimal requirement for that is that the complexity of the code does not add to the complexity of the problem it is trying to solve. Otherwise over time the existing code becomes the actual problem. At the moment you understand the problem and your code. This is the best time to improve the code to really express the problem. In a few months time nobody will remember why the code is that way it was written and to change it will then require a much bigger effort.

Reply:
*pinches bridge of nose*  okay, fred.

Reply:
I moved processSegues:... into the method which processes the storyboard.  The result is stored in the identifierToSegueMap  (this is the segue id).  My goal is the same as yours.  I prefer maintainable code, but I also like to take an iterative approach and optimize it over time.  I really think that's the only difference.   I, however, concede to your judgment as you're the GUI maintainer.  I won't merge without your approval.